Sources/Jobs/JobOptions.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import struct Foundation.Date
+
+/// Job options
+public struct JobOptions: Sendable {
+    /// When to execute the job
+    public let delayUntil: Date?
```

Comment:
I'm of the camp of making things a `let` by default, even in structs, but think should make this a `var` instead

Reply:
> I'm of the camp of making things a `let` by default, even in structs, but think should make this a `var` instead

Should I change it to a var instead? I was under the impression, a user would set `delayUntil` once when pushing a job and would not need to mutate it after.